tests/test_schema.py

```diff
@@ -736,52 +758,144 @@ def test_function_ignore_parameters():
+    :param d: This is a list parameter
+    """
+    return a, b, c, d
+
+
+def test_function_ignore_parameter_descriptions():
+    rf = ReferenceSchema(function_ignore_parameter_descriptions)
+    rf.remove_parameter_descriptions()
+    assert function_ignore_parameter_descriptions.schema.to_json() == rf.schema
+    assert function_ignore_parameter_descriptions.schema.to_json(SchemaType.TUNE) == rf.tune_schema
+    assert function_ignore_parameter_descriptions.tags == []
+
+
+#####################################
+#  Test ignore_function_description #
+#####################################
+
+
+@GPTEnabled(ignore_function_description=True)
+def function_ignore_function_description(a: int, b: str, c: bool = False, d: list[int] = [1, 2, 3]):
+    """
+    This is a test function.
+
+    :param a: This is a parameter
+    :param b: This is another parameter
+    :param c: This is a boolean parameter
+    :param d: This is a list parameter
+    """
+    return a, b, c, d
+
+
+def test_function_ignore_function_description():
+    rf = ReferenceSchema(function_ignore_function_description)
+    rf.get_function().pop("description")
+    assert function_ignore_function_description.schema.to_json() == rf.schema
+    assert function_ignore_function_description.schema.to_json(SchemaType.TUNE) == rf.tune_schema
+    assert function_ignore_function_description.tags == []
+
+
+###############################
+#  Test ignore_all_parameters #
+###############################
+
+
+@GPTEnabled(ignore_all_parameters=True)
+def function_ignore_all_parameters(a: int, b: str, c: bool = False, d: list[int] = [1, 2, 3]):
+    """
+    This is a test function.
+
+    :param a: This is a parameter
+    :param b: This is another parameter
+    :param c: This is a boolean parameter
+    :param d: This is a list parameter
+    """
+    return a, b, c, d
+
+
+def test_function_ignore_all_parameters():
+    rf = ReferenceSchema(function_ignore_all_parameters)
+    rf.get_parameters().pop("properties")
+    rf.get_parameters().pop("required")
+    assert function_ignore_all_parameters.schema.to_json() == rf.schema
+    assert function_ignore_all_parameters.schema.to_json(SchemaType.TUNE) == rf.tune_schema
+    assert function_ignore_all_parameters.tags == []
+
+
 ###############################
 #  Test global configuration  #
 ###############################
 
 
-def test_global_configuration_ignore_parameters():
-    # Change the global configuration
-    tool2schema.CONFIG.ignore_parameters = ["b", "c"]
+@pytest.fixture
+def global_config() -> tool2schema.config.Config:
+    yield tool2schema.CONFIG
 
-    # We have to re-define the function in this scope
-    # to use the updated configuration
-    @GPTEnabled
-    def _function(a: int, b: str, c: bool = False, d: list[int] = [1, 2, 3]):
-        """
-        This is a test function.
+    # Ensure the global configuration is reset to the default after the test
+    tool2schema.CONFIG.reset_default()
```

Comment:
Why do you reset after `yield` instead of just reset then `return`?

Reply:
It is the pytest way of running code after a test has been executed
From [stackoverflow](https://stackoverflow.com/questions/22627659/run-code-before-and-after-each-test-in-py-test):

```python
@pytest.fixture(autouse=True)
def run_around_tests():
    # Code that will run before your test, for example:
    files_before = # ... do something to check the existing files
    # A test function will be run at this point
    yield
    # Code that will run after your test, for example:
    files_after = # ... do something to check the existing files
    assert files_before == files_after
```